src/Algebra/Graph.hs

```diff
@@ -462,6 +462,25 @@ hasVertex x = foldg False (==x) (||) (||)
 hasEdge :: Ord a => a -> a -> Graph a -> Bool
 hasEdge u v = (edge u v `isSubgraphOf`) . induce (`elem` [u, v])
 
+-- | Check if a graph contains a given loop.
+-- Complexity: /O(s)/ time.
+--
+-- @
+-- hasSelfLoop x 'empty'            == False
+-- hasSelfLoop x ('vertex' z)       == False
+-- hasSelfLoop x ('edge' x x)       == True
+-- hasSelfLoop x                    == hasEdge x x
```

Comment:
I think these examples do not line up in Haddock. Could you please check? You might need to insert a few spaces here and there.